indico_toolkit/indico_wrapper/dataset.py

```diff
@@ -68,42 +69,73 @@ def create_empty_dataset(
         """
         return self.client.call(CreateEmptyDataset(dataset_name, dataset_type))
 
+    def create_dataset(self, filepaths: List[str], dataset_name: str) -> Dataset:
+        dataset = self.client.call(
+            CreateDataset(
+                name=dataset_name,
+                files=filepaths,
+            )
+        )
+        self.dataset_id = dataset.id
+        return dataset
+
     def create_large_doc_dataset(
         self,
         dataset_name: str,
         filepaths: List[str],
-        batch_size: int = 3,
-        verbose: bool = True,
-    ) -> Dataset:
+        num_threads: int = 3,
+        upload_batch_size: int = 3,
+        process_batch_size: int = 5,
+        max_download_checks: int = 25,
+    ):
         """
-        Create a dataset and batch document uploads to not overload queue/services
+        Uses a threadpool to upload large documents one at a time
+        and add them to a new dataset
 
         Args:
             dataset_name (str): create a name for the dataset
             filepaths (List[str]): files you want to upload
-            batch_size (int, optional): number of files to process at a time. Defaults to 3.
-            verbose (bool, optional): print updates as each batch processes. Defaults to True.
+            num_threads (int, optional): number of threads to use for file upload. Defaults to 3.
+            upload_batch_size (int, optional): number of files to upload at a time. Defaults to 3.
+            process_batch_size (int, optional): number of files to process at a time. Defaults to 5.
+            max_download_checks (int, optional): number of times to poll for uploads to complete.
+                                                 Necessary because documents can get stuck in a DOWNLOADING
+                                                 state. Defaults to 10 (~1 minute)
         """
         fp = FileProcessing(filepaths)
         dataset_id = self.create_empty_dataset(dataset_name).id
-        number_of_batches = int(math.ceil(len(filepaths) / batch_size))
-        for batch_fpaths in fp.batch_files(batch_size):
-            dataset = self.add_files_to_dataset(dataset_id, batch_fpaths)
-            if verbose:
-                number_of_batches -= 1
-                print(
-                    f"Finished batch at {datetime.now().time()}: {number_of_batches} remaining"
+
+        # Upload using an executor - take advantage of multiple uploaders on the platform
+        results = self._upload_threaded(fp, num_threads, upload_batch_size)
+        print("Uploaded Doucments")
+
+        add_files_batch_size = 100
```

Comment:
should this be from one of the kwargs? why 100 hardcoded?

Reply:
missed this, I'll move that out